### PR TITLE
feat(ingest): refactor mlModel grouping and add browsepaths

### PR DESCRIPTION
**Pull request by kevinhu**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/ml/metadata/MLModelGroupProperties.pdl [hunk 1]
```diff
@@ -31,15 +31,4 @@ record MLModelGroupProperties includes CustomProperties {
...
-  /**
-   * Models in the group
-   */
-  @Relationship = {
-    "/*": {
-      "name": "Includes",
-      "entityTypes": [ "mlModel" ]
-    }
-  }
-  models: array[Urn] = [ ]
```
Comment (dexter-mh-lee) [inline metadata-models/src/main/pegasus/com/linkedin/ml/metadata/MLModelGroupProperties.pdl:44]: Is this backward compatible?

File: metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/models.py [hunk 2]
```diff
@@ -333,6 +384,24 @@ def get_model_wu(
...
+
+        model_group_names = sorted(
+            [self.group_arn_to_name[x] for x in model_group_arns]
+        )
+        model_group_urns = [
+            builder.make_ml_model_group_urn("sagemaker", x, self.env)
+            for x in model_group_names
+        ]
+
+        model_browsepaths = [
```
Comment (dexter-mh-lee) [inline metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/models.py:401]: Does this cover the case where groups may not exist?

File: metadata-models/src/main/pegasus/com/linkedin/ml/metadata/MLModelGroupProperties.pdl (hunk 1 above)
Comment (kevinhu) [reply metadata-models/src/main/pegasus/com/linkedin/ml/metadata/MLModelGroupProperties.pdl:44]: Note needed – model groups haven't been released yet!

File: metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/models.py (hunk 2 above)
Comment (kevinhu) [reply metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/models.py:401]: Will add!

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**